### PR TITLE
[TGBOT-240] Extract in memory repository implementation in separate module

### PR DESCRIPTION
**Pull request by Sav4ykVladzislau**
https://schneidervs.atlassian.net/browse/TGBOT-240

### CODE REVIEW

**Review by schneidervs — APPROVED** [review]

**Review by AlexanderYarokhovich — APPROVED** [review]

**Review by yurasava — DISMISSED** [review]

**Review by Artem-Metlushko — APPROVED** [review]

**Review by schneidervs — APPROVED** [review]

**Review by yurasava — DISMISSED** [review]

File: persistence-core/src/main/java/com/tutrit/persistence/core/persistence/CarPersistence.java [hunk 1]
```diff
@@ -4,6 +4,5 @@
 
 public interface CarPersistence {
     Car save(Car car);
-
-    Car findById(String carId);
+    Car findById(String id);
```
Comment (max-java) [inline persistence-core/src/main/java/com/tutrit/persistence/core/persistence/CarPersistence.java:7]: I suggest to leave `carId` as a name for id

File: persistence-core/src/main/java/com/tutrit/persistence/core/persistence/EngineerPersistence.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.tutrit.persistence.core.persistence;
+
+import com.tutrit.persistence.core.bean.Engineer;
+
+public interface EngineerPersistence {
+    Engineer save(Engineer engineer);
+
+    Engineer findById(String id);
```
Comment (max-java) [inline persistence-core/src/main/java/com/tutrit/persistence/core/persistence/EngineerPersistence.java:8]: I suggest engineerId as name for variable.

**Review by max-java — CHANGES_REQUESTED** [review]
ptal

File: rest-api-service/pom.xml [hunk 3]
```diff
@@ -60,4 +54,4 @@
         </plugins>
     </build>
 
-</project>
+   </project>
```
Comment (schneidervs) [inline rest-api-service/pom.xml:57]: This is unnecessary.

**Review by schneidervs — APPROVED** [review]

**Review by yurasava — APPROVED** [review]

**Review by AlexanderYarokhovich — APPROVED** [review]

**Review by Artem-Metlushko — DISMISSED** [review]

File: in-memory-repository/src/main/java/com/tutrit/persistence/InMemoryCarPersistence.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Component
+public class InMemoryCarPersistence implements CarPersistence {
+    public static final Logger logger = LoggerFactory.getLogger("CarPersistence logger");
+    private static final List<Car> cars = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (Sav4ykVladzislau) [reply in-memory-repository/src/main/java/com/tutrit/persistence/InMemoryCarPersistence.java:15]: To solve this problem, a task was created:
https://schneidervs.atlassian.net/browse/TGBOT-303

File: in-memory-repository/src/main/java/com/tutrit/persistence/InMemoryCustomerPersistence.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.UUID;
+
+
+@Component
+public class InMemoryCustomerPersistence implements CustomerPersistence {
+    public static final Logger logger = LoggerFactory.getLogger("CustomerPersistence logger");
+    private final static HashSet<Customer> customers = new HashSet<>();
```
[the remark this answers is not in the document]
Comment (Sav4ykVladzislau) [reply in-memory-repository/src/main/java/com/tutrit/persistence/InMemoryCustomerPersistence.java:14]: To solve this problem, a task was created:
https://schneidervs.atlassian.net/browse/TGBOT-303

**Review by max-java — APPROVED** [review]
lgfm

File: rest-api-service/pom.xml [hunk 6]
```diff
@@ -60,4 +57,18 @@
...
+                <dependency>
+                    <groupId>com.tutrit</groupId>
+                    <artifactId>in-memory-repository</artifactId>
+                    <version>0.1.0-SNAPSHOT</version>
+                    <scope>compile</scope>
+                </dependency>
+            </dependencies>
+        </profile>
+    </profiles>
+
```
[the remark this answers is not in the document]
Comment (Sav4ykVladzislau) [reply rest-api-service/pom.xml]: I remove this profile and i created task https://schneidervs.atlassian.net/browse/TGBOT-306

**Review by yurasava — APPROVED** [review]

**Review by schneidervs — APPROVED** [review]

**Review by Artem-Metlushko — APPROVED** [review]

**Review by AlexanderYarokhovich — DISMISSED** [review]

**Comment by max-java** [comment]
Please, rebase on main and squash.

**Review by max-java — APPROVED** [review]
I hope resolved discussions done properly , new changes lgfm. 
let's merge and see the result

**Review by Artem-Metlushko — APPROVED** [review]

**Review by AlexanderYarokhovich — APPROVED** [review]

**Review by yurasava — APPROVED** [review]

**Review by schneidervs — APPROVED** [review]

**Pull request merged**